osc/osc_test.go

```diff
@@ -316,7 +316,9 @@ func TestReadPaddedString(t *testing.T) {
 		s   string // resulting string
 	}{
 		{[]byte{'t', 'e', 's', 't', 's', 't', 'r', 'i', 'n', 'g', 0, 0}, 12, "teststring"},
-		{[]byte{'t', 'e', 's', 't', 0, 0, 0, 0}, 8, "test"},
+		{[]byte{'t', 'e', 's', 't', 'e', 'r', 's', 0}, 8, "testers"},
+		{[]byte{'t', 'e', 's', 't', 's', 0, 0, 0}, 8, "tests"},
+		{[]byte{'t', 'e', 's', 't'}, 4, "test"},
```

Comment:
Replace these two with:

```go
{[]byte{'t', 'e', 's', 0, 0, 0, 0, 0}, 4, "tes", nil}, // OSC uses null terminated strings
{[]byte{'t', 'e', 's', 't'}, 0, "", io.EOF}, // if there is no null byte at the end, it doesn't work.
```

Reply:
> ```go
> {[]byte{'t', 'e', 's', 0, 0, 0, 0, 0}, 4, "tes", nil}, // OSC uses null terminated strings
> ```

Surely this one should be:
> ```go
> {[]byte{'t', 'e', 's', 0}, 4, "tes", nil}, // OSC uses null terminated strings
> ```

Which is what I've committed, as the null terminator takes the total string length to 4 bytes so no padding required.

Reply:
Travis seems happy with my correction too @chabad360 unless you think I've missed something?

Reply:
Reason why I had included the extra 4 bytes, was just to ensure that they are ignored.

Reply:
Makes sense, but the length should still be four as it's output not input length. I've added it and another couple of separate tests.